python_modules/dagster/dagster/components/cli/list.py

```diff
@@ -157,6 +159,17 @@ def list_definitions_impl(
             repository_origin = get_repository_python_origin_from_cli_opts(python_pointer_opts)
             recon_repo = recon_repository_from_origin(repository_origin)
             repo_def = recon_repo.get_definition()
+            tree = check.not_none(repo_def.get_component_tree())
+
+            try:
+                defs = tree.load_defs_at_path(path) if path else tree.load_defs()
```

Comment:
defs are already loaded at this point - utility of this flag for now is just limiting output, not making the command faster